### PR TITLE
signInWithToken

### PR DESCRIPTION
**Pull request by bdlukaa**
Fixes https://github.com/supabase-community/supabase-flutter/issues/5

See https://github.com/supabase/gotrue/pull/189

### CODE REVIEW

File: lib/src/gotrue_api.dart [hunk 1]
```diff
@@ -139,6 +141,66 @@ class GoTrueApi {
...
 
+  /// Logs in an existing user using an id token from a provider.
+  ///
+  /// `idToken` is the user id token with nonce
+  ///
+  /// `nonce` is the unhashed nonce in id token
+  ///
+  /// `provider` is the id provider. Can be `apple`, `google`, `azure` or
+  /// `facebook`
+  Future<GotrueSessionResponse> signInWithToken(
```
Comment (dshukertjr) [inline lib/src/gotrue_api.dart:152]: I wonder if we should use the same method name as the Javascript SDK. Was there a particular reason why you changed the method name here?

https://github.com/supabase/gotrue-js/blob/master/src/GoTrueApi.ts#L227

**Review by dshukertjr — COMMENTED** [review]
I would love to actually test this out later when I have some time, but overall looks amazing! Looking forward to the complete PR!

File: lib/src/gotrue_api.dart (hunk 1 above)
Comment (bdlukaa) [reply lib/src/gotrue_api.dart:152]: I didn't know there was a javascript function for this... https://github.com/supabase/gotrue-js/issues/169 is still open

Comment (DanMossa) [reply lib/src/gotrue_api.dart:152]: It was actually merged in. I think the issue is still just open

**Comment by bdlukaa** [comment]
Closing in favor or #61

**Pull request closed without merging**